### ISSUE TITLE
ReflectedDecoratorAttribute gives mangled name in MethodInfo

### ISSUE THREAD

**Issue by chkn** [fable-compiler/Fable#2723]
### Description

Following the steps in https://fable.io/blog/2021/2021-09-17-JS-decorators-Fable-3-3.html to use `ReflectedDecoratorAttribute`, the `MethodInfo` contains the mangled name of the function.

### Repro code

The REPL doesn't seem to have the decorator attributes, but you can repro by following these steps.

1. Create the following files:

`Foo.fsproj`
```xml
<Project Sdk="Microsoft.NET.Sdk">
    <PropertyGroup>
        <OutputType>Exe</OutputType>
        <TargetFramework>net6.0</TargetFramework>
    </PropertyGroup>
    <ItemGroup>
        <Compile Include="FableDecorator.fs" />
        <PackageReference Include="Fable.Core" Version="3.6.1" />
    </ItemGroup>
</Project>
```

`FableDecorator.fs`
```fsharp
open Fable.Core.JS

type FooAttribute() =
    inherit ReflectedDecoratorAttribute()
    override _.Decorate(fn, info) =
        printfn $"Decorating function {info.Name}"
        fn

[<Foo>]
let ``foo bar baz`` () = ()
```

2. `dotnet fable Foo.fsproj`
3. `mv FableDecorator.fs.js FableDecorator.fs.mjs`
4. `node FableDecorator.fs.mjs`

### Expected and actual results

Expected to print: "Decorating function foo bar baz"
Actual output: "Decorating function foo$0020bar$0020baz"

### Related information

* Fable version: 3.7.0-beta-008
* Operating system: macOS 10.15.7

**Comment by alfonsogarciacaro** [comment]
Yes, this is an unfortunate design choice from my part. Using the mangled name was intended because Fable.Lit uses it to identify the member in the JS module when doing hot reloading. I now realized I should have used the original name and provide the mangled name through a special Fable extension. I just need to find a way to do it without breaking Fable.Lit 😅

**Comment by chkn** [comment]
Ah makes sense. I'm using it in an experiment to see if I can bring my unit test framework to Fable. The name is just used to print the name of the test, so it's not a big deal right now. I could maybe contribute a fix if it progresses past an experiment.

The best thing would be if it were possible to decorate a whole module and get something like a `System.Type` with all the `MethodInfo`s, but that's a separate thing :)
